test/e2e/framework/framework.go

```diff
@@ -185,23 +185,23 @@ func (f *Framework) BeforeEach() {
 			config.ContentType = TestContext.KubeAPIContentType
 		}
 		f.ClientSet, err = clientset.NewForConfig(config)
-		Expect(err).NotTo(HaveOccurred())
```

Comment:
Isn't `ExpectNoError(err)` without an additional message just as bad as `Expect(err).NotTo(HaveOccurred())` regarding the information contained in the output when the check fails?

Reply:
@pohly Sorry for my late response.
ExpectNoError() can output `err` to log as
```
2162 func ExpectNoError(err error, explain ...interface{}) {
2163         ExpectNoErrorWithOffset(1, err, explain...)
2164 }
2165
2166 // ExpectNoErrorWithOffset checks if "err" is set, and if so, fails assertion while logging the error at "offset" levels above its caller
2167 // (for example, for call chain f -> g -> ExpectNoErrorWithOffset(1, ...) error would be logged for "f").
2168 func ExpectNoErrorWithOffset(offset int, err error, explain ...interface{}) {
2169         if err != nil {
2170                 Logf("Unexpected error occurred: %v", err)
2171         }
2172         ExpectWithOffset(1+offset, err).NotTo(HaveOccurred(), explain...)
2173 }
```
So the method seems better than `Expect(err).NotTo(HaveOccurred())` which doesn't output it, I think.

Reply:
I don't think that this is correct. `Expect(err).NotTo(HaveOccurred())` does print the error as can be seen in the description of https://github.com/kubernetes/kubernetes/issues/34059, but often the error doesn't have enough information ("timed out waiting for the condition" - which condition?!).

The additional information is what has to be provided via the optional messages parameter, regardless whether `NotTo` or ` `ExpectNoError` are used.